src/services/RPCSelector.ts

```diff
@@ -42,17 +41,26 @@ class RPCSelector {
     return this.getProviderWithHighestBlockNumber(providerComparands);
   }
 
-  private async isProviderUpToDate(url: string): Promise<boolean> {
-    try {
-      const provider = providers.getDefaultProvider(url);
-      const block = <{ timestamp: number }>await Promise.race([provider.getBlock('latest'), this.timeout()]);
-      return isTimestampMoreRecentThan(block.timestamp, this.config.maxTimestampDiff / 1000);
-    } catch {
-      return false;
-    }
+  private getProvidersTimestamps(): Promise<ComparandWithTimestamp>[] {
```

Comment:
I think here you should return an array of Promises, in this case you are returning the promises resolved, so maybe the Promise.all() will not work as expected.

Reply:
It is returning an array of promises. They will be only executed in `Promise.all`